BUILD

```diff
@@ -0,0 +1,112 @@
+load("@rules_rust//rust:defs.bzl", "rust_library", "rust_proc_macro", "rust_binary")
+
+rust_library(
+    name = "autocxx",
+    edition = "2021",
+    srcs = glob(["src/**/*.rs"]),
+    data = glob(["README.md"]),
+    deps = [
+        "//third-party:cxx",
+        "//third-party:moveit",
+    ],
+    visibility = ["//visibility:public"],
+    proc_macro_deps = [
+	":autocxx-macro",
+        "//third-party:aquamarine",
+    ],
+)
+
+rust_proc_macro(
+    name = "autocxx-macro",
+    srcs = glob(["macro/src/**/*.rs"]),
+    edition = "2021",
+    visibility = ["//visibility:public"],
+    deps = [
+	":autocxx-parser",
+        "//third-party:proc-macro-error",
+        "//third-party:proc-macro2",
+        "//third-party:quote",
+        "//third-party:syn",
+    ],
+)
+
+rust_library(
+    name = "autocxx-parser",
+    srcs = glob(["parser/src/**/*.rs"]),
+    edition = "2021",
+    deps = [
+        "//third-party:log",
+        "//third-party:proc-macro2",
+        "//third-party:quote",
+        "//third-party:serde",
+        "//third-party:thiserror",
```

Comment:
Thanks for this PR.

However, I hadn't realized that bazel support would require hard-coding all these dependencies. That's not OK with me - the `Cargo.toml` files need to be the sole source of truth - can you instead add a script or tool to generate these `BUILD` files based on the `Cargo.toml`?

More generally, I'm concerned that the project structure is not static enough to maintain a duplicate set of build instructions.

Reply:
Thank you for the review. Yes, it was possible to drop the hard-coding deps in `BUILD` by [all_crate_deps](https://bazelbuild.github.io/rules_rust/crate_universe.html#all_crate_deps). I missed this utility.